libraries/USBHost/src/hidescriptorparser.cpp

```diff
@@ -1088,6 +1088,15 @@ void ReportDescParserBase::PrintItemTitle(uint8_t prefix) {
         } // switch (**pp & (TYPE_MASK | TAG_MASK))
 }
 
+#pragma GCC diagnostic push // Available since GCC 4.6.4

```

Comment:
you could use the comment to tell gcc that the fall through is on purpose
```
// FALLTHRU 
```

Reply:
The comment method is fragile.  See:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78497

Correct fixes:
* [ ] For C++17 and higher, use [`[[fallthrough]]`](https://en.cppreference.com/w/cpp/language/attributes/fallthrough) attribute
* [ ] For GCC, use `__attribute__(( fallthrough ))`
* [ ] For CLANG, use `[[clang::fallthrough]]` attribute

If you know this code will only be compiled with GCC, then we can mark it with `__attribute__(( fallthrough ))`.

Else, I recommend deferring on this (using pragma push/pop) so as to give it separate consideration.  Likely could look at defining a macro such as ADAFRUIT_FALLTHROUGH, that conditionally defines to C++17, GCC, or CLANG as available.  But I recommend a separate PR for that....

Reply:
More importantly ... I don't think it's right to simply disable the warnings.  It's very possible that one or more of the implicit fallthrough are **_not_** intentional.  However, if the code is modified to explicitly mark it as an expected fallthrough, then it can hide a bug.

Because I did not have time to review the functions to this depth, I did not add any new fallthrough indicators.

Reply:
please just use the fallthru comment, I have no issues with it so far with other project. We could try another method when gcc cannot dectect this particular project should it arise.

Reply:
I have the following in my `platform.local.txt`:

```
compiler.c.extra_flags= -save-temps=obj
compiler.cpp.extra_flags= -save-temps=obj
compiler.S.extra_flags= -save-temps=obj
```

See the [bug](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78497): the comment method is not a proper solution ... it's a legacy hack that even GCC doesn't recommend.  The attribute usage is the correct solution.  (`-save-temps=obj` is super-useful when trying to understand why macros aren't working, and tracking down many tricky types of bugs...)

Therefore, GCC cannot detect the comments, and GCC recommends the use of attributes.

Reply:
Ah, OK, let's use the  `__attribute__(( fallthrough ))` I have no problems with it, pretty much the same as fallthru comment.

Reply:
@henrygab can you either add the fallthru attribute or revert those changes. Either way, the `#pragma` should be removed.